adding-more-native-to-your-react-native-app.md

```diff
@@ -0,0 +1,186 @@
+`react-native` command-line tool, in that it will actually edit your Xcode project on-disk. One limitation with this
+approach was that the tool doesn’t know about **native** transitive dependencies, and as such in those cases you would
+still have to perform the manual approach for those dependencies.
+
+### What problems exist with this approach?
+
+1. In an ideal scenario, you would have to perform 2 steps to manually integrate a single native package, at worst
+   including additional build settings and multiplied by the number of **native** transitive dependencies the native
+   package has. Even in the automated case, the inconsistency in handling **native** transitive dependencies isn’t the
+   greatest developer-experience.
+1. Needing to understand the build settings of Xcode and its underlying tooling. This isn’t necessarily a bad thing,
+   things _can_ and _will_ break and it’s good to know how things work under the hood, but _having_ to do this each time
+   even when on the happy path isn’t ideal.
+1. Other than bookkeeping, there’s no easy way to track which build settings originated from what dependency, which
+   becomes a maintenance burden when you want to upgrade or remove dependencies.
+
+## Alternative and the future of React Native linking
+
+This is where relying on dependency managers comes in. Respective ecosystems already have dependency management
+solutions that are able to deal with all of the above, so in the spirit of standing on the shoulders of giants, let’s
+leverage those domain specific tools instead!
+
+In the case of iOS, the popular option is [CocoaPods](https://cocoapods.org)–which given a manifest that describes a
+project’s dependencies, similarly to a `package.json` file, will perform all of the aforementioned work _and_ do so in a
+maintainable manner. For instance, it is able to resolve and install transitive dependencies like any other dependency,
+it encapsulates dependency specific build settings in `xcconfig` files, and does so with minimal _one-time_ changes to
+your app’s Xcode project.
+
+Seeing as sooner or later most people will have to deal with linking native packages and the different ways in which you
+had to maintain your app’s Xcode project, the React Native maintainers decided to make utilizing CocoaPods the default
+since [version 0.60](https://facebook.github.io/react-native/blog/2019/07/03/version-60#cocoapods-by-default). (Gradle,
+for Android apps, was already the default.)
+
+So, great–now there’s a single place where you manage your native dependencies! But given this standardization, we can
+take it one step further and automate the process. 🐢s all the way down!
+
+### Auto-linking
+
+Enter auto-linking, which is a new addition to the `react-native` command-line tool that discovers packages that have
+native requirements and offloads the responsibility of configuring the app project to the dependency manager for the
+respective platform.
```

Comment:
@bartolkaruza You mentioned that the Android side of auto-linking does not yet support transitive dependencies. Should I mention that hereabout? And can you give me a summary of why it doesn’t do that yet? (I assumed that through Gradle that would be handled transparently.)

Reply:
Hmm, unsure what you mean here, let’s catchup IRL at the meetup.